### ISSUE TITLE
`blankLinesAtStartOfScope` rule in disabled list is not being disabled.

### ISSUE THREAD

**Issue by PompeiaPaetenari** [nicklockwood/SwiftFormat#507]
In our `swiftformat.config`, we have `--disable blankLinesAtStartOfScope` set.  However, in the latest version (I tested it in 41.2 when trying to format a file), the formatter is adding these newlines again.  I have double-checked the casing and even used a different swiftformat.cfg file without the rule and am seeing that newlines still get added at the start of scope.

When I run it on version 40.13, I do not have the problem and the rule behaves as expected.  No newlines are added or removed as compared to before.

For a quick test, I have the following two config files:
swiftformat.cfg (this one removed the newlines, as the rule is not disabled):
```
--allman true
--indent tab
--tabwidth 3

--disable andOperator,blankLinesAroundMark,consecutiveSpaces,elseOnSameLine,fileHeader
--enable isEmpty
```

swiftformat.config (this adds newlines, despite the rule being in disabled list):
```
--commas always
--decimalgrouping 3
--indent 2
#--maxwidth 140
--semicolons never
--wraparguments beforefirst
--wrapcollections beforefirst

--disable redundantReturn,blankLinesAtStartOfScope,andOperator
```

I run the commands in order as:
```
$ swiftformat --config ../swiftformat.cfg .
$ swiftformat --config swiftformat.config .
```

I then compare the before and after files with Fork (or just look via `git diff` or in editor of choice, etc.) to see a ton of newlines added at the start of scope for each function, enum, computed variable, etc.  I would like to be able to update and use the maxwidth 140 option to go with swiftlint, but the formatter with this bug in it would cause nearly every file in our code base to be changed.
When I rerun the two config files in the same steps (top, then bottom), but using the 40.13 version of swiftformat, the newlines are not added.

**Comment by nicklockwood** [comment]
The name is a bit misleading, but the `blankLinesAtStartOfScope` is responsible for *removing* blank lines at the start of a scope, not adding them, so the blank lines are being added by a different rule.

I can't be sure which rule is causing this, based on your description, but you might want to try disabling the following rules:

blankLinesBetweenScopes
blankLinesAroundMark

If that doesn't help, you could try using the `--rules` option to explicitly opt-in to the rules that you want to use instead of using `--disable` to filter out the rules you don't want.

**Comment by nicklockwood** [comment]
Actually, I see the problem.

You have `--allman true` in your first config file, which moves the opening braces `{` onto the next line. You don't have `--allman` in your second config, and it defaults to false, so this moves the `{` back onto the previous line, leaving a blank line.

This blank line would be removed by the `blankLinesAtStartOfScope` rule, but you've disabled it.

**Comment by PompeiaPaetenari** [comment]
So the issue still stands of how the behaviour changed from 40.13 where it not only moved the brace up, but cleared out the newline.  I saw nothing in the changelog that reflects any such alteration to the behaviour, hence posting the bug for clarification.

I depend on your tool to see code the way it is easier for me to read, with allman, but then change it back to commit for the rest of the team.  They have already said they do not want the existing state of newlines after functions to be altered, as there are special cases for them, so I cannot change the team-wide rules to remove them... but rather leave them in their existing state (there or not there) and that is the reason they have the rule disabled.

**Comment by nicklockwood** [comment]
@PompeiaPaetenari you're right, this is a regression. I was under the impression it always worked this way, but it seems not. I believe this was a side-effect of

- Fixed bug where a multiline comments before an opening brace could result in corrupted output

In the 0.41.0 release. I'll fix it in the next release.

**Comment by nicklockwood** [comment]
@PompeiaPaetenari fixed in version 0.42.0.
